### ISSUE TITLE
wcstok.c: 3 copyrights, but only 2 detected?

### ISSUE THREAD

**Issue by ened** [nexB/scancode-toolkit#130]
Given I have a file header like this:

```
/*-
 * Copyright (c) 1998 Softweyr LLC.  All rights reserved.
 *
 * strtok_r, from Berkeley strtok
 * Oct 13, 1998 by Wes Peters <wes@softweyr.com>
 *
 * Copyright (c) 1988, 1993
 *  The Regents of the University of California.  All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 * 1. Redistributions of source code must retain the above copyright
 *    notices, this list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notices, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 * 3. Neither the name of the University nor the names of its contributors
 *    may be used to endorse or promote products derived from this software
 *    without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY SOFTWEYR LLC, THE REGENTS AND CONTRIBUTORS
 * ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
 * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
 * PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL SOFTWEYR LLC, THE
 * REGENTS, OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
 * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
 * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
 * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
 * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
 * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
```

Then, `scancode`, when started like this:

```
scancode -c ./libc/upstream-freebsd/lib/libc/string/wcstok.c
```

Outputs this:

```
{
  "scancode_notice": "Generated with ScanCode and provided on an \"AS IS\" BASIS, WITHOUT WARRANTIES\nOR CONDITIONS OF ANY KIND, either express or implied. No content created from\nScanCode should be considered or used as legal advice. Consult an Attorney\nfor any legal advice.\nScanCode is a free software code scanning tool from nexB Inc. and others.\nVisit https://github.com/nexB/scancode-toolkit/ for support and download.", 
  "scancode_version": "1.3.8", 
  "resource_count": 1, 
  "results": [
    {
      "location": "./libc/upstream-freebsd/lib/libc/string/wcstok.c", 
      "copyrights": [
        {
          "statements": [
            "Copyright (c) 1998 Softweyr LLC."
          ], 
          "start_line": 2, 
          "end_line": 2
        }, 
        {
          "statements": [
            "Copyright (c) 1988, 1993 The Regents of the University of California."
          ], 
          "start_line": 7, 
          "end_line": 8
        }
      ]
    }
  ]
}
```

Whereas I would expect to find a notice about "Wes Peters" or is that wrong?

**Comment by pombredanne** [comment]
if I `export SCANCODE_DEBUG_COPYRIGHT=1` to visualize the parse tree, here is what is found there:
```
DEBUG:cluecode.copyrights:CopyrightDetector:parse tree: (S
  strtok_r,/NN
  from/NN
  (NAME Berkeley/NNP strtok/NN Oct/NNP)
  (YR-RANGE 13,/CD 1998/YR)
  by/BY
  (AUTHOR (NAME2 (NAME Wes/NNP Peters/NNP) <wes@softweyr.com>/EMAIL)))
```
This would not be detected as a copyright, but is detected as an author. The authors are not yet returned in the results, but they should. 

Here this is a case that is clear where `Wes Peters` is the author and not the copyright holder.

So net-net we should return the authors in scans

**Comment by pombredanne** [comment]
I am keeping this open until author are exposed as a scan

**Comment by ened** [comment]
That would be a new node in the JSON export, same level like "copyrights", and should get a cmd line option, too?

**Comment by pombredanne** [comment]
@ened wrote:
>That would be a new node in the JSON export, same level like "copyrights", and should get a cmd line option, too?

Yes, exactly.

**Comment by pombredanne** [comment]
actually the authors are always detected together with the copyrights internally, but they are just not yet returned.

**Comment by pombredanne** [comment]
As you can see here:
https://github.com/nexB/scancode-toolkit/blob/a95929579bcad25e80db3416f647562e8a69121d/src/scancode/api.py#L58
`for copyrights, _, _, _, start_line, end_line in detect_copyrights(location):` really means:
 `copyrights list, authors list, years list, holders list, start line, end line`
as returned by https://github.com/nexB/scancode-toolkit/blob/a95929579bcad25e80db3416f647562e8a69121d/src/cluecode/copyrights.py#L63

**Comment by pombredanne** [comment]
Authors are exposed in scans alright (and have been for a while)
This has been fixed in develop and I added more test in #790
Thank you!
